From 01775609525b12de752a2f7ebac88baee9b041d3 Mon Sep 17 00:00:00 2001
From: Carlos Garnacho <carlosg@gnome.org>
Date: Fri, 29 Oct 2021 16:46:18 +0200
Subject: [PATCH 3/3] inputMethod: Do not reset invisible preedit on focus_out

If the preedit is in invisible state, the last preedit string that
was sent around is already null, so there is no need to clear the preedit
in that case.

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2011>
---
 js/misc/inputMethod.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/misc/inputMethod.js b/js/misc/inputMethod.js
index 71dff84f5..8c3cd61ae 100644
--- a/js/misc/inputMethod.js
+++ b/js/misc/inputMethod.js
@@ -167,7 +167,7 @@ class InputMethod extends Clutter.InputMethod {
         if (this._context)
             this._context.focus_out();
 
-        if (this._preeditStr) {
+        if (this._preeditStr && this._preeditVisible) {
             // Unset any preedit text
             this.set_preedit_text(null, 0, this._preeditCommitMode);
             this._preeditStr = null;
-- 
2.33.1

